plugins/version-control-systems/git/src/main/kotlin/Git.kt

```diff
@@ -285,7 +311,14 @@ class Git internal constructor() : VersionControlSystem(GitCommand) {
             revision
         }
 
-    private fun updateSubmodules(workingTree: WorkingTree) {
+    /**
+     *  Initialize, update, and clone all the submodules in a working tree.
+     *
+     *  If [recursive] is set to true, then the operations are not only performed on the
+     *  submodules in the top-level of the working tree, but also on the submodules of the submodules, and so on.
+     *  If [recursive] is set to false, only the submodules on the top-level are initialized, updated, and cloned.
```

Comment:
Please also document the `historyDepth` parameter (and consider swapping the order of the `recursive` and `historyDepth` parameters).

I believe the docs could be a bit more concise, like:

> Initialize and / or update the submodules in [workingTree], limiting commit history to the number of [historyDepth]. If [recursive] is true, submodules are initialized / updated recursively. Otherwise only top-level ones are processed.